containers/base_rpm.go

```diff
@@ -0,0 +1,44 @@
+package containers
+
+import "dagger.io/dagger"
+
+const RPMMacros = `
+%_signature gpg
+%_gpg_path /root/.gnupg
+%_gpg_name Grafana
+%_gpgbin /usr/bin/gpg2
+%__gpg_sign_cmd %{__gpg} gpg \
+	--batch --yes --no-armor --pinentry-mode loopback \
+	--passphrase-file /root/.rpmdb/passkeys/grafana.key \
+	--no-secmem-warning -u "%{_gpg_name}" -sbo %{__signature_filename} \
+	%{?_gpg_digest_algo:--digest-algo %{_gpg_digest_algo}} %{__plaintext_filename}
+`
+
+func RPMContainer(d *dagger.Client, opts *SignOpts) *dagger.Container {
+	container := FPMContainer(d).
+		WithExec([]string{"apt-get", "update"}).
+		WithExec([]string{"apt-get", "install", "-yq", "rpm"})
+	if !opts.Sign {
+		return container
+	}
+	return container.
+		WithExec([]string{"apt-get", "install", "-yq", "gnupg2"}).
+		WithNewFile("/root/.rpmmacros", dagger.ContainerWithNewFileOpts{
+			Permissions: 0400,
+			Contents:    RPMMacros,
+		}).
+		WithNewFile("/root/.rpmdb/privkeys/grafana.key", dagger.ContainerWithNewFileOpts{
+			Permissions: 0400,
+			Contents:    opts.GPGPrivateKey,
+		}).
+		WithNewFile("/root/.rpmdb/pubkeys/grafana.key", dagger.ContainerWithNewFileOpts{
+			Permissions: 0400,
+			Contents:    opts.GPGPublicKey,
+		}).
+		WithNewFile("/root/.rpmdb/passkeys/grafana.key", dagger.ContainerWithNewFileOpts{
+			Permissions: 0400,
+			Contents:    opts.GPGPassphrase,
+		}).
```

Comment:
This could be a security risk; we need to make sure that there's no way that the gpg private key or passphrase could be logged into the terminal.

For this, we can use a dagger Secret. There's an example of using a secret in some gcloud file or publish.go I think.

Basically you'll create a secret, populate it with a value, then mount it as an environment variable, then you'll probably have to `[]string{"/bin/sh", "-c", "cat $ENV >  /root/.rpmdb/privkeys/grafana.key"}`

Reply:
Done! Take a look and let me know!